Dockerfile

```diff
@@ -0,0 +1,161 @@
+#
+# STEP 4
+# Follow the instructions on how to start the Oppia server.
+#
+##########################################################################
+######################## TO START THE OPPIA SERVER #######################
+##########################################################################
+#
+# STEP 1
+# Start bash in the Docker image:
+#
+#   docker run -u 0 -it -p 8181:8181 {some_name}:latest /bin/bash
+#   ex: docker run -u 0 -it -p 8181:8181 oppia_image:latest /bin/bash
+#
+# If Docker outputs an error, run:
+#   docker ps
+# This displays a repository name and tag name. Now run:
+#   docker run -u 0 -it -p 8181:8181 {repository}:{tag} /bin/bash
+#
+# STEP 2
+# Now you should have a new terminal prompt "root@...". Run the start.sh
+# script:
+#
+#   bash scripts/start.sh
+#
+# Runtime: 10-20 minutes
+# The script opens a server at localhost:8181. After the terminal prints
+# "INFO ... http://0.0.0.0:8181" or "+ 27 hidden modules", open 
+# localhost:8181 in your local computer browser. If the Oppia server does
+# not load, restart this step.
+#
+##########################################################################
+########################## TO RUN FRONTEND TESTS #########################
+##########################################################################
+#
+# STEP 1
+# Start bash in the Docker image (follow STEP 1 above)
+#
+# STEP 2
+# Run the run_frontend_tests.sh script:
+#
+#   bash scripts/run_frontend_tests.sh
+#
+# Runtime: 3-7 minutes
+# If this outputs an error ("failed before timeout of 2000ms"), continue
+# to STEP 3.
+# If this runs correctly (displays "SUCCESS"), do not go onto STEP 3.
+#
+# STEP 3
+# Compile the frontend tests and then run the tests:
+#
+#   ./node_modules/typescript/bin/tsc --project .
+#   ./node_modules/karma/bin/karma start ./core/tests/karma.conf.ts
+#
+# If this outputs an error, try STEP 2 again.
+# If this runs correctly, you will see a SIGKILL at the end. That is okay.
+
+##########################################################################
+
+FROM ubuntu:latest
+
+# Install packages needed in Dockerfile
+RUN apt-get update && \
+   apt-get install -y sudo && \
+   apt-get install -y vim && \
+   apt-get install -y wget && \
+   apt-get install -y nodejs && \
+   apt-get install -y npm && \
+   apt-get install -y yes && \
+   apt-get clean
+
+# Create oppia directory in Docker container
+RUN mkdir /home/oppia
+
+# Install prerequisites. The yes package responds "yes" to all prompts.
+COPY ./scripts/install_prerequisites.sh /home/oppia/scripts/
+RUN yes | bash /home/oppia/scripts/install_prerequisites.sh
+
+# Fix certificate issues
+RUN apt-get update && \
+   apt-get install ca-certificates-java && \
+   apt-get clean && \
+   update-ca-certificates -f
+
+# Setup JAVA_HOME
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
+ENV PATH $PATH:$JAVA_HOME/bin
+RUN export JAVA_HOME
+
+# Install dumb-init (Signal handling of SIGINT/SIGTERM/SIGKILL etc.)
```

Comment:
Can you elaborate a bit more on this comment? I.e. explain exactly what dumb-init is needed for and what would go wrong without it. It's OK to include details, don't just say "etc.".

This is useful context for future developers.

Reply:
Good point. Done.